scripts/upgrade_2024_01_10.py

```diff
@@ -0,0 +1,322 @@
+        assert not l1_token_bridge.isDepositsEnabled()
+
+    # L1 Bridge has old implementation
+    l1_token_bridge_implementation_address_before = l1_token_bridge_proxy.proxy__getImplementation()
+    assert l1_token_bridge_implementation_address_before == L1_OPTIMISM_TOKENS_BRIDGE_IMPL, "Old address is incorrect"
+
+    # L1 Bridge doesn't have version before update
+    try:
+        l1_token_bridge.getContractVersion()
+    except VirtualMachineError:
+        pass
+
+    # Upgrade LidoLocator implementation
+    lido_locator_impl_before = lido_locator_proxy.proxy__getImplementation()
+    assert lido_locator_impl_before == LIDO_LOCATOR_IMPL, "Old address is incorrect"
+
+    # Multisig hasn't been assigned as deposit enabler
+    assert not l1_token_bridge.hasRole(DEPOSITS_ENABLER_ROLE, L1_EMERGENCY_BRAKES_MULTISIG)
+
+def pause_deposits():
+    if not network_name() in ("mainnet-fork"):
+        return
+
+    network.gas_price("2 gwei")
+
+    account_wht_eth = "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2"
+    accountWithEth = accounts.at(account_wht_eth, force=True)
+    accountWithEth.transfer(AGENT, "2 ethers")
+
+    l1_token_bridge = interface.L1LidoTokensBridge(L1_OPTIMISM_TOKENS_BRIDGE)
+    agent = accounts.at(AGENT, force=True)
+    l1_token_bridge.disableDeposits({"from": agent})
+    assert not l1_token_bridge.isDepositsEnabled()
+
+def resume_deposits():
+    if not network_name() in ("mainnet-fork"):
+        return
+
+    network.gas_price("2 gwei")
+
+    account_wht_eth = "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2"
+    accountWithEth = accounts.at(account_wht_eth, force=True)
+    accountWithEth.transfer(AGENT, "2 ethers")
+
+    l1_token_bridge = interface.L1LidoTokensBridge(L1_OPTIMISM_TOKENS_BRIDGE)
+    agent = accounts.at(AGENT, force=True)
+    l1_token_bridge.enableDeposits({"from": agent})
+    assert l1_token_bridge.isDepositsEnabled()
+
+
+def check_post_upgrade_state(vote_tx):
+    l1_token_bridge_proxy = interface.OssifiableProxy(L1_OPTIMISM_TOKENS_BRIDGE)
+    l1_token_bridge = interface.L1LidoTokensBridge(L1_OPTIMISM_TOKENS_BRIDGE)
+    lido_locator_proxy = interface.OssifiableProxy(LIDO_LOCATOR)
+
+    # L1 Bridge has new implementation
+    l1_token_bridge_implementation_address_after = l1_token_bridge_proxy.proxy__getImplementation()
+    assert (
+        l1_token_bridge_implementation_address_after == L1_OPTIMISM_TOKENS_BRIDGE_IMPL_NEW
+    ), "New address is incorrect"
+
+    # update L1 Bridge to 2 version
+    assert l1_token_bridge.getContractVersion() == 2
+
+    # LidoLocator has new implementation
+    lido_locator_impl_after = lido_locator_proxy.proxy__getImplementation()
+    assert lido_locator_impl_after == LIDO_LOCATOR_IMPL_NEW, "New LidoLocator address is incorrect"
+
+    # Multisig has been assigned as deposit enabler
+    assert l1_token_bridge.hasRole(DEPOSITS_ENABLER_ROLE, L1_EMERGENCY_BRAKES_MULTISIG)
+    # TODO: check multisig can renounce the role
```

Comment:
either add the check (make tx to renounce the role from msig) or remove this TODO. I think either option is fine